src/utils/helperFunctions/metadataFilters.ts

```diff
@@ -1,25 +1,46 @@
 import { DataStreamWithRoutes } from "src/utils/api/dataStreams";
-import { Route } from "src/utils/api/routes";
+import { RouteWithPermissions } from "src/utils/api/routes";
 
 export const getDataStreamIds = (data: DataStreamWithRoutes[]) => {
-  const dataStreamIds: string[] = data.map(
-    (el: DataStreamWithRoutes) => el.datastream.name
-  );
-  return dataStreamIds;
+  return data.map((el: DataStreamWithRoutes) => el.datastream.name);
 };
 
 export const getDataRoutes = (
   data: DataStreamWithRoutes[],
   dataStreamName: string,
   forUpload: boolean = false
 ): string[] => {
-  const dataStream: DataStreamWithRoutes | undefined = data.find(
+  const dataStream = data.find(
     (el: DataStreamWithRoutes) => el.datastream.name == dataStreamName
   );
 
   if (!dataStream) return [];
 
-  const routeNames = dataStream.routes.map((r: Route) => r.name);
+  const routeNames = dataStream.routes.map((r: RouteWithPermissions) => r.name);
+
+  if (routeNames.length > 1 && !forUpload) return ["All", ...routeNames];
+
+  return routeNames;
+};
+
+export const getDataStreamIdsWriteOnly = (data: DataStreamWithRoutes[]) => {
```

Comment:
A lot of these filter functions look pretty similar.  Wonder if you can consolidate to just one for filtering on just the write permission flag and return the whole object, and then can do the mapping to get names in the component itself.

Reply:
These functions are used to create the options for the datastream and route drop downs. So it is less about filtering and more about mapping. They are used in a few places on the Portal, so I don't really want to have that mapping logic in all of those places. But I agree it could be less repeated logic. refactoring now.